cdap-examples/LogAnalysis/src/main/java/co/cask/cdap/examples/loganalysis/LogAnalysisApp.java

```diff
@@ -165,4 +180,89 @@ public void centers(HttpServiceRequest request, HttpServiceResponder responder,
+
+    @UseDataSet(REQ_COUNT_STORE)
+    private TimePartitionedFileSet reqCountStore;
+
+    /**
+     * Handler which lists all the different time partitions available in the {@link LogAnalysisApp#REQ_COUNT_STORE}
+     * {@link TimePartitionedFileSet}
+     */
+    @Path(REQUEST_COUNTER_PARTITIONS_PATH)
+    @GET
+    public void getRequestFilesetPartitions(HttpServiceRequest request, HttpServiceResponder responder) {
+
+      // get all the existing paritions
+      Set<TimePartitionDetail> partitionsByTime = reqCountStore.getPartitionsByTime(0, Long.MAX_VALUE);
+
+      SortedSet<String> formattedTimes = new TreeSet<>();
+      for (TimePartitionDetail timePartitionDetail : partitionsByTime) {
+        String partitionTime = SHORT_DATE_FORMAT.format(new Date(timePartitionDetail.getTime()));
+        formattedTimes.add(partitionTime);
+      }
+      responder.sendJson(HttpURLConnection.HTTP_OK, formattedTimes);
+    }
+
+    /**
+     * Handler which reads all the parts files from a given partition in {@link LogAnalysisApp#REQ_COUNT_STORE}
+     * {@link TimePartitionedFileSet} and send it as a string.
+     * Note: We make an assumption here that the contents for partitions in the tpfs for this example is not very huge.
+     * This method of serving contents is not ideal for large contents.
+     */
+    @Path(REQUEST_FILE_CONTENT_PATH)
+    @POST
+    public void getRequestFilesetContents(HttpServiceRequest request, HttpServiceResponder responder) {
+
+      String partition = GSON.fromJson(Charsets.UTF_8.decode(request.getContent()).toString(),
+                                       JsonObject.class).get(REQUEST_FILE_PATH_HANDLER_KEY).getAsString();
+      long partitionKey = 0;
+      try {
+        partitionKey = SHORT_DATE_FORMAT.parse(partition).getTime();
+      } catch (ParseException e) {
+        responder.sendError(HttpURLConnection.HTTP_BAD_REQUEST, "Failed to parse the given string to a timestamp");
+        return;
+      }
+
+      final Location location = reqCountStore.getPartitionByTime(partitionKey).getLocation();
+      if (location == null) {
+        responder.sendError(HttpURLConnection.HTTP_NOT_FOUND, "No files for the given date time string");
+        return;
+      }
+
+      Map<String, Integer> requestCountsMap = Maps.newHashMap();
+      try {
+        for (Location file : location.list()) {
+          if (file.getName().startsWith("part")) {
+            String line;
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(file.getInputStream(),
+                                                                                  Charsets.UTF_8))) {
+              while ((line = reader.readLine()) != null) {
+                int idx = line.indexOf(":");
+                requestCountsMap.put(line.substring(0, idx), Integer.parseInt(line.substring(idx + 1)));
+              }
+            } catch (IOException e) {
```

Comment:
I think its cleaner as I want to differentiate between error in reading file and error in accessing the partition dir. Don't you agree ?